File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.cpp
#include <bits/stdc++.h>
using namespace std;

string solution(string s, string skip, int index) {
    bool possible[26] ;
    string answer = "";
    memset(possible, 0, sizeof(possible));
    for(char c : skip)
        possible[c % 26] = 1 ;

    for(char c : s)
    {
        int now = index ;
        for(int i = 1 ; i <= now ; i++)
        {
            if(possible[(c + i) % 26]) now++ ;
        }
        now = now % 26 ;
        answer += (c + now) > 'z' ? c + now - 26 : c + now;
    }
        
    return answer;
}

File: 프로그래머스/unrated/155652. 둘만의 암호/둘만의 암호.py
from string import ascii_lowercase

def solution(s, skip, index):
    answer = ''
    word = set(ascii_lowercase)
    word -= set(skip)
    word = sorted(word)

    dic_alpha = {alpha:idx for idx, alpha in enumerate(word)}
    
    for i in s:
        answer += word[(dic_alpha[i] + index) % len(word)]
    return answer